include/envoy/thread_local/thread_local.h

```diff
@@ -76,19 +76,14 @@ class Slot {
   virtual void set(InitializeCb cb) PURE;
 
   /**
-   * UpdateCb takes the current stored data, and must return the same data. The
-   * API was designed to allow replacement of the object via this API, but this
-   * is not currently used, and thus we are removing the functionality. In a future
-   * PR, the API will be removed.
-   *
-   * TLS will run the callback and assert the returned returned value matches
-   * the current value.
+   * UpdateCb takes is passed a shared point to the current stored data. Use of
+   * this API is deprecated; please use TypedSlot::runOnAllThreads instead.
```

Comment:
Can we just make this private and use a friend declaration or something like that?

Reply:
protected, yeah. I had been thinking about that but got lazy. Done now. Doing so made it necessary to move the testing to the TypedSlot wrapper, but i think that makes sense.

It also exposed some users of Slot::runOnAllThreads that I had somehow missed earlier, so those needed to be cleaned up.